File: clients/go/examples/double/double.go
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// double is a vega example demonstrating writing to multiple writers.
package main

import (
	"flag"
	"log"
	"math"
	"multiscope/clients/go/scope"
)

var (
	httpPort = flag.Int("http_port", scope.DefaultPort, "http port")
	local    = flag.Bool("local", true, "open the port to local connection only")
)

func main() {
	flag.Parse()
	if err := scope.Start(*httpPort, *local); err != nil {
		log.Fatal(err)
	}
	ticker, err := scope.NewTicker("ticker", nil)
	if err != nil {
		log.Fatal(err)
	}
	// Create two writers. Each writer will have its own plot.
	wSin, err := scope.NewScalarWriter("sin", ticker.Path())
	if err != nil {
		log.Fatal(err)
	}
	wCos, err := scope.NewScalarWriter("cos", ticker.Path())
	if err != nil {
		log.Fatal(err)
	}
	for {
		if err = ticker.Tick(); err != nil {
			break
		}
		t := float64(ticker.CurrentTick()) / 100
		if err = wSin.Write(map[string]any{
			"sa": math.Sin(t),
			"sb": math.Sin(t * 2),
			"sc": math.Sin(t / 2),
		}); err != nil {
			break
		}
		if err = wCos.Write(map[string]any{
			"ca": math.Cos(t),
			"cb": math.Cos(t * 2),
			"cc": math.Cos(t / 2),
		}); err != nil {
			break
		}
	}
	log.Fatal(err)
}

File: clients/python/orig_source/examples/double.py
# Copyright 2023 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Double is an example illustrating how to use two writers."""

import math

from absl import app

import multiscope
from multiscope.examples import examples


def main(_):
  multiscope.start_server()
  wsin = multiscope.ScalarWriter("sin")
  wsin.set_history_length(400)
  wcos = multiscope.ScalarWriter("cos")
  wcos.set_history_length(400)
  for time in examples.step():
    wsin.write({
        "sin": math.sin(time * 0.01),
    })
    wcos.write({
        "cos": math.cos(time * 0.01),
    })


if __name__ == "__main__":
  app.run(main)
